util/src/main/java/org/folio/metastorage/util/OaiRecord.java

```diff
@@ -0,0 +1,25 @@
+package org.folio.metastorage.util;
+
+public class OaiRecord<T> {
+  String datestamp;
```

Comment:
Maybe. I'll try that.

Reply:
It will be impossible to determine whether this was full-date- time or just date with this. It's +1 second or +1 day to get next until. So it will have to do (for now).

Reply:
Good point. I guess the simplest fix is to put an enum Granularity {SECONDS, DAYS}? Otherwise one could try with LocalDate and LocalDateTime hidden behind Temporal and isSupported but that seems very convoluted.

Reply:
It will not be simpler. I can make an OaiDate with two members. This can be used as the member for OaiRecord . I can *also* be nice wrapper for handling the until/from that is used in the server, so focus all OAI date handling there.

Reply:
Sounds good to me.